ksqldb-common/src/main/java/io/confluent/ksql/schema/ksql/SystemColumns.java

```diff
@@ -65,14 +93,26 @@ public static boolean isPseudoColumn(final ColumnName columnName) {
   }
 
   public static Set<ColumnName> pseudoColumnNames() {
-    return PSEUDO_COLUMN_NAMES;
+    return getPseudoColumnsFromVersion(CURRENT_PSEUDOCOLUMN_VERSION_NUMBER);
   }
 
   public static boolean isSystemColumn(final ColumnName columnName) {
     return systemColumnNames().contains(columnName);
   }
 
   public static Set<ColumnName> systemColumnNames() {
-    return SYSTEM_COLUMN_NAMES;
+    return SYSTEM_COLUMN_NAMES_CURRENT;
+  }
+
+  //cache version numbers we have used, otherwise create as needed
```

Comment:
I'm not convinced this is optimization (the cache) is necessary. As long as this method is not on the critical path (i.e., we don't call it once per message processed), then we don't need to over-optimize at the expense of additional code complexity. 

(That said, this type of caching is a neat trick! File it away for the future.)